lib/web_ui/test/engine/pointer_binding_test.dart

```diff
@@ -246,80 +244,80 @@ void testMain() {
+    expect(events[0].changedTouches![0].identifier, equals(102));
+    expect(events[0].changedTouches![0].client.x, equals(220));
+    expect(events[0].changedTouches![0].client.y, equals(221));
+    expect(events[0].changedTouches![1].identifier, equals(103));
+    expect(events[0].changedTouches![1].client.x, equals(222));
+    expect(events[0].changedTouches![1].client.y, equals(223));
 
     event = expectCorrectType(context.primaryUp(clientX: 300, clientY: 301));
     expect(event.type, equals('touchend'));
-    expect(event.changedTouches.length, equals(1));
-    expect(event.changedTouches[0].identifier, equals(1));
-    expect(event.changedTouches[0].client.x, equals(300));
-    expect(event.changedTouches[0].client.y, equals(301));
+    expect(event.changedTouches!.length, equals(1));
+    expect(event.changedTouches![0].identifier, equals(1));
+    expect(event.changedTouches![0].client.x, equals(300));
+    expect(event.changedTouches![0].client.y, equals(301));
 
     events = expectCorrectTypes(context.multiTouchUp(<_TouchDetails>[
       _TouchDetails(pointer: 104, clientX: 320, clientY: 321),
       _TouchDetails(pointer: 105, clientX: 322, clientY: 323),
     ]));
     expect(events.length, equals(1));
     expect(events[0].type, equals('touchend'));
-    expect(events[0].changedTouches.length, equals(2));
-    expect(events[0].changedTouches[0].identifier, equals(104));
-    expect(events[0].changedTouches[0].client.x, equals(320));
-    expect(events[0].changedTouches[0].client.y, equals(321));
-    expect(events[0].changedTouches[1].identifier, equals(105));
-    expect(events[0].changedTouches[1].client.x, equals(322));
-    expect(events[0].changedTouches[1].client.y, equals(323));
+    expect(events[0].changedTouches!.length, equals(2));
+    expect(events[0].changedTouches![0].identifier, equals(104));
+    expect(events[0].changedTouches![0].client.x, equals(320));
+    expect(events[0].changedTouches![0].client.y, equals(321));
+    expect(events[0].changedTouches![1].identifier, equals(105));
+    expect(events[0].changedTouches![1].client.x, equals(322));
+    expect(events[0].changedTouches![1].client.y, equals(323));
 
     events = expectCorrectTypes(context.multiTouchCancel(<_TouchDetails>[
       _TouchDetails(pointer: 104, clientX: 320, clientY: 321),
       _TouchDetails(pointer: 105, clientX: 322, clientY: 323),
     ]));
     expect(events.length, equals(1));
     expect(events[0].type, equals('touchcancel'));
-    expect(events[0].changedTouches.length, equals(2));
-    expect(events[0].changedTouches[0].identifier, equals(104));
-    expect(events[0].changedTouches[0].client.x, equals(320));
-    expect(events[0].changedTouches[0].client.y, equals(321));
-    expect(events[0].changedTouches[1].identifier, equals(105));
-    expect(events[0].changedTouches[1].client.x, equals(322));
-    expect(events[0].changedTouches[1].client.y, equals(323));
+    expect(events[0].changedTouches!.length, equals(2));
```

Comment:
Maybe extract these to a local variable, so we don't need that many bangs?

`List<Whatever> changedTouches = events[0].changedTouches!` ?

Reply:
There's much more repetition going on than the bangs. I'd prefer to keep the nullability concern separate. Would you mind if that clean-up is done separately?